resource-managers/kubernetes/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala

```diff
@@ -69,33 +77,48 @@ private[spark] class KubernetesClusterSchedulerBackend(
 
   override def start(): Unit = {
     super.start()
-    val keyValuePairs = Map("num-executors" -> getInitialTargetExecutorNumber(sc.conf),
-      "image" -> sparkImage,
-      "state" -> JobState.SUBMITTED)
-    try {
-      logInfo(s"Creating Job Resource with name. $podPrefix")
-      sparkJobResource.createJobObject(podPrefix, keyValuePairs)
-    } catch {
-      case e: SparkException =>
-        logWarning(s"SparkJob object not created. ${e.getMessage}")
+    logInfo(s"Updating Job Resource with name. $jobObjectName")
+    Try(sparkJobResource
+      .updateJobObject(jobObjectName, JobState.SUBMITTED.toString, "/spec/state")) match {
+      case Success(_) => startWatcher()
+      case Failure(e: SparkException) =>
+        logWarning(s"SparkJob object not updated. ${e.getMessage}")
       // SparkJob should continue if this fails as discussed on thread.
       // TODO: we should short-circuit on things like update or delete
     }
     createExecutorPods(getInitialTargetExecutorNumber(sc.getConf))
   }
 
+  private def startWatcher(): Unit = {
+    sparkJobResource.watchJobObject() onComplete {
+      case Success(w: WatchObject) if w.`type` == "DELETED" =>
+        isObjectDeleted = true
+        logInfo("TPR Object deleted. Cleaning up")
+        stop()
+      case Success(_: WatchObject) => throw new SparkException("Unexpected response received")
+      case Failure(e: Throwable) => throw new SparkException(e.getMessage)
```

Comment:
Yes. This call back is triggered when the source from the server is exhausted (failure case) or the resource being watched is deleted (success case).

On the success case, it stops the spark job by cleaning up the pods.

Pending question is: what is the right thing to do if the callback returns with a failure, right now I defaulted to throwing an error because this means the state of the SparkJob resource is now unknown and the watcher has stopped.